web/src/components/ResourceCard/TestCard.tsx

```diff
@@ -13,19 +13,23 @@ import ResourceCardSummary from './ResourceCardSummary';
 import useRuns from './useRuns';
 
 interface IProps {
+  onEdit(id: string, lastRunId: number, type: ResourceType): void;
   onDelete(id: string, name: string, type: ResourceType): void;
   onRun(test: TTest, type: ResourceType): void;
   onViewAll(id: string, type: ResourceType): void;
   test: TTest;
 }
 
-const TestCard = ({onDelete, onRun, onViewAll, test}: IProps) => {
+const TestCard = ({onEdit, onDelete, onRun, onViewAll, test}: IProps) => {
   const queryParams = useMemo(() => ({take: 5, testId: test.id}), [test.id]);
   const {isCollapsed, isLoading, list, onClick} = useRuns<TTestRun, {testId: string}>(
     useLazyGetRunListQuery,
     queryParams
   );
 
+  const canEdit = test.summary.runs > 0;
```

Comment:
I think it would make sense to calculate these two things at the model level `web/src/models/Test.model.ts` if we are going to use it in different places.

Reply:
One thing that I did here to fit both Tests and Transactions was to add a `hasRuns` field into the `TestSummary` model. Does it make sense?